src/ts/ConvivaAnalytics.ts

```diff
@@ -144,19 +123,18 @@ export class ConvivaAnalytics {
    * relies on the players source we can't ensure that all metadata attributes are present at session creation.
    * Therefore it could be that there will be a 'ContentMetadata created late' issue after conviva validation.
    *
-   * @param assetName Will be used as contentMetadata.assetName if no source was loaded before. This overrides the
-   *                  source assetName. If no source was loaded and no assetName is present this method will throw an
-   *                  error.
+   * If no source was loaded and no assetName was set via updateContentMetadata this method will throw an error.
    */
-  public initializeSession(assetName?: string): void {
-    // This could be called before source loaded.
-    // Without setting the asset name on the content metadata when initializing the session the SDK will throw errors.
-    if (!this.player.getSource() && !assetName) {
-      throw('AssetName is missing. Provide assetName attribute or load player source first');
+  public initializeSession(): void {
+    if (this.isValidSession()) {
+      this.logger.consoleLog('There is already a session running.', Conviva.SystemSettings.LogLevel.WARNING);
+      return;
     }
 
-    if (assetName) {
-      this.metadataOverrides.assetName = assetName;
+    // This could be called before source loaded.
+    // Without setting the asset name on the content metadata when initializing the session the SDK will throw errors.
```

Comment:
`the session the SDK will throw errors` something is wrong here

Reply:
It's just strange for Austrian brains to see no commas in such a sentence. I think it is valid english syntax though.

Maybe we could do with some commas though?